PluginCore/PluginCore/Controls/MemoryMessageBox.cs

```diff
@@ -0,0 +1,124 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Windows.Forms;
+using PluginCore.Controls;
+using PluginCore.Localization;
+
+namespace PluginCore.PluginCore.Controls
+{
+    public enum MemoryMessageBoxResult
+    {
+        Yes,
+        YesToAll,
+        No,
+        NoToAll,
+        Cancel
+    }
+
+    public partial class MemoryMessageBox : SmartForm
+    {
+        readonly Dictionary<MemoryMessageBoxResult, Button> resultToButton;
+        MemoryMessageBoxResult lastResult = MemoryMessageBoxResult.Cancel;
+        public MemoryMessageBoxResult Result { get; set; } = MemoryMessageBoxResult.Cancel;
+
+        public MemoryMessageBox()
+        {
+            Owner = (Form) PluginBase.MainForm;
+            Font = PluginBase.MainForm.Settings.DefaultFont;
+            InitializeComponent();
+            ApplyLocalizedTexts();
+            resultToButton = new Dictionary<MemoryMessageBoxResult, Button>
+            {
+                {MemoryMessageBoxResult.Yes, yesButton},
+                {MemoryMessageBoxResult.YesToAll, yesToAllButton},
+                {MemoryMessageBoxResult.No, noButton},
+                {MemoryMessageBoxResult.NoToAll, noToAllButton},
+                {MemoryMessageBoxResult.Cancel, cancelButton}
+            };
+        }
+
+        /// <summary>
+        /// Applies the localized texts to the form
+        /// </summary>
+        private void ApplyLocalizedTexts()
+        {
+            yesButton.Text = TextHelper.GetString("Label.Yes");
+            yesToAllButton.Text = TextHelper.GetString("Label.YesToAll");
+            noButton.Text = TextHelper.GetString("Label.No");
+            noToAllButton.Text = TextHelper.GetString("Label.NoToAll");
+            cancelButton.Text = TextHelper.GetString("Label.Cancel");
+            dontAsk.Text = TextHelper.GetString("Label.DontAskAgain");
+        }
+
+        public MemoryMessageBoxResult ShowDialog(string text, string caption)
+        {
+            label.Text = text;
+            Text = caption;
+            return ShowDialog();
+        }
+
+        public new MemoryMessageBoxResult ShowDialog()
+        {
+            Result = MemoryMessageBoxResult.Cancel;
+            if (dontAsk.Checked)
```

Comment:
I think this is dangerous - "don't ask" should only apply to a specific operation, not any operation using this dialog.

Reply:
Good otherwise.

Reply:
false by default. But I think I can add the ability to hide this control.

Reply:
Ah sorry I thought it was shared across all the instances -  so that will work only if you keep the same dialog instance?

Reply:
Yes